src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs

```diff
@@ -17,115 +19,413 @@
 namespace Microsoft.PowerFx
 {
     /// <summary>
-    /// Result of binding an expression. 
+    /// Holds work such as parsing, binding, error checking done on a single expression. 
+    /// Different options require different work. 
+    /// Tracks which work is done so that it is not double repeated.
     /// </summary>
     public class CheckResult : IOperationStatus
     {
-        /// <summary> 
-        /// Return type of the expression. Null if type can't be determined. 
+        #region User Inputs
+        // Captured from user, set once via a Set*() function. 
+
+        /// <summary>
+        /// The source engine this was created against. 
+        /// This is critical for calling back to populate the rest of the results. 
         /// </summary>
-        public FormulaType ReturnType { get; set; }
+        private readonly Engine _engine;
+       
+        // The raw expression test. 
+        private string _expression;
+        
+        private ParserOptions _parserOptions;
+
+        // Information for binding. 
+        private bool _setBindingCalled;
+        internal ReadOnlySymbolTable _symbols; // can be null
+        private VersionHash _symbolHash; // hash of _symbols at time of assignment
+
+        #endregion
+
+        [Obsolete("Use public constructor")]
+        internal CheckResult()
+        {
+        }

```

Comment:
Is it time to remove this one?

Reply:
Added for PAClient compat to avoid an internal breaking change. Kept internal to mitigate use. Can remove after PAClient sweeps to remove Obsoletes. .